src/agentic/actor_agents.py

```diff
@@ -1474,6 +1473,9 @@ def next_turn(self, request: str | Prompt, request_context: dict = {},
                     request_id=request_id,
                 )
             )
+            # There's some logical issue where we have inserted 'thread_id' into the request_context, but
```

Comment:
@drbrady8800 this came up in some testing, where `thread_context.thread_id` inside a tool function was None when I expected a value. Seems like we had a case where the thread ID wouldn't get passed down into the agent loop properly.